source/s3_default_meta_request.c

```diff
@@ -105,25 +106,24 @@ struct aws_s3_meta_request *aws_s3_meta_request_default_new(
     }
 
     meta_request_default->content_length = (size_t)content_length;
-    meta_request_default->request_type = request_type;
-
-    /* Try to get operation name.
-     * When internal aws-c-s3 code creates a default meta-request,
-     * a valid request_type is always passed in, and we can get its operation name.
-     * When external users create a default meta-request, they may have provided
-     * operation name in the options. */
-    const char *operation_name = aws_s3_request_type_operation_name(request_type);
-    if (operation_name[0] != '\0') {
-        meta_request_default->operation_name = aws_string_new_from_c_str(allocator, operation_name);
-    } else if (options->operation_name.len != 0) {
+
+    if (request_type == AWS_S3_REQUEST_TYPE_UNKNOWN) {
+        /* use name to get type */
         meta_request_default->operation_name = aws_string_new_from_cursor(allocator, &options->operation_name);
+        meta_request_default->request_type = aws_s3_request_type_from_operation_name(options->operation_name);
+    } else {
+        /* use type to get name */
+        meta_request_default->request_type = request_type;
+        meta_request_default->operation_name =
+            aws_string_new_from_c_str(allocator, aws_s3_request_type_operation_name(request_type));
```

Comment:
worth to think about make some generic operation name as a static string, so that we don't make too many copy/paste of the same string over and over again? May not be a bottle neck, but seems like a waste for cases like a lot small files.

Reply:
did it